### PR TITLE
Refactor handlers

### PR DESCRIPTION
**Pull request by charles1046**
I refactored handlers and tried to manually resolve merge conflict.
But due to file creations, I'm unable to fully resolve the conflict.

### CODE REVIEW

**Comment by charles1046** [comment]
`src/http/cookies.c`: Typo. Not really a conflict.
`src/facebooc_server.c`: Handlers are moved to separated source files.
`src/server.c`: Handler of handlers is moved to a separated source file.

**Comment by charles1046** [comment]
Fixed 404 after liking.

**Comment by charles1046** [comment]
@charles1046
[Refactor database finalize statement](https://github.com/25077667/facebooc/pull/24/commits/2cc6027dd55be8cf7679705999e933e48c9f5dd1)

File: src/server.c [hunk 1]
```diff
@@ -114,36 +116,28 @@ static void serverDelFd(Server* server, int fd) {
...
 		if(errno == EAGAIN) {
 			resetOneShot(server->priv, fd);
 		}
 		else {
 			fprintf(stderr, "error: read failed\n");
 		}
 	}
-	else if(nread > 0) {
-		buff[nread] = '\0';
+	if(n_read > 0) {
```
Comment (25077667) [inline src/server.c:130]: Use else if here.

File: src/handler/handler.c [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	add_handler(handlers, "profile", profile);
+	add_handler(handlers, "dashboard", dashboard);
+	add_handler(handlers, "", home);
+	add_handler(handlers, "static", static_handler);
+
+	add_callback_handler(handlers);
+}
+
+Response* handle_request(Hash_map* handlers, Request* request) {
+	char* handling_type = get_handling_type(request);
```
Comment (25077667) [inline src/handler/handler.c:64]: I prefer to use "routing path".

**Review by 25077667 — CHANGES_REQUESTED** [review]
Please see the comments. Thanks!

**Comment by charles1046** [comment]
Updated.

File: src/handler/handler.c [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+#include "handler/handlers/like.h"
+#include "handler/handlers/login.h"
+#include "handler/handlers/logout.h"
+#include "handler/handlers/post.h"
+#include "handler/handlers/profile.h"
+#include "handler/handlers/search.h"
+#include "handler/handlers/signup.h"
+#include "handler/handlers/unlike.h"
+
+static inline void add_handler(Hash_map* handlers, char* handling_name, Handler handler) {
```
Comment (25077667) [inline src/handler/handler.c:28]: We should not add handlers here.
Each server has a set of handlers, and the facebooc server is a derived class of a server.
We could register some handlers into a base class, but not modify it directly.

Please refer to Open-closed principle.

**Review by 25077667 — CHANGES_REQUESTED** [review]
Please see the comments.
Thanks.

**Review by 25077667 — COMMENTED** [review]
Viewed.

**Pull request closed without merging**